src/components/common/needsShutdown.jsx

```diff
@@ -85,6 +85,26 @@ export function needsShutdownSpice(vm) {
     return vm.hasSpice !== vm.inactiveXML.hasSpice;
 }
 
+export function needsShutdownVnc(vm) {
+    function find_vnc(v) {
+        return v.displays && v.displays.find(d => d.type == "vnc");
+    }
+
+    const active_vnc = find_vnc(vm);
+    const inactive_vnc = find_vnc(vm.inactiveXML);
+
+    if (inactive_vnc) {
+        if (!active_vnc)
+            return true;
+        if (inactive_vnc.port != -1 && active_vnc.port != inactive_vnc.port)
```

Comment:
I don't understand the -1 comparison. If you change from "-1" (which means "automatic/dynamic" presumably?) to a static port, that should also count as a change.

Reply:
Isn't that what this does? When the inactive XML has a static port (!= -1) and it is different from what the running VM uses, it's a change.

The logic here is meant to only catch things that Cockpit can do itself. Not sure if that makes total sense. So for example, when the user removed VNC completely with virt-xml, the Cockpit will not tell people to shutdown. People can keep using their VNC undisturbed until they shutdown by themselves. Does that make sense?

Reply:
It appeared to me that you can go from static to dynamic port in cockpit, i.e. by unticking the "[ ] Use static port" check box. Even if you do it on the command line, cockpit should always react to it -- so telling you that the VM needs to be restarted for the change to take action seems correct. Either way this extra comparison feels like an unnecessary complication and restriction?